structures.h

```diff
@@ -0,0 +1,7 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include "structures.c"

```

Comment:
#include "structures.c"  is obsolete. And probably wrong....

Reply:
Replace contents of .h and .c files....

Reply:
In c files the implemtntation of functions ar placed

Reply:
Also add header guurades here

Reply:
It is better also to split the structures each in different .h. and .c files. Yo can do it afterward...